extensions/amp-story/0.1/amp-story.js

```diff
@@ -1434,6 +1449,52 @@ export class AmpStory extends AMP.BaseElement {
     this.switchTo_(dev().assertElement(this.pages_[0].element).id);
   }
 
+  /** @return {!NavigationState} */
+  getNavigationState() {
+    return this.navigationState_;
+  }
+
+
+  /**
+   * Add page to back of pages_ array
+   * @param {!./amp-story-page.AmpStoryPage} page
+   */
+  addPage(page) {
+    this.pages_.push(page);
+
+    if (page.isAd()) {
+      this.adPages_.push(page);
+    }
+  }
+
+  /**
+   * Insert a new page in navigation flow by changing the attr pointers
+   * on amp-story-page elements
+   * @param {string} currentPageId
+   * @param {string} pageToBeInsertedId
+   */
+  insertPage(currentPageId, pageToBeInsertedId) {
+    // TODO(ccordry): make sure this method moves to PageManager when implemented
+    const pageToBeInserted = this.getPageById_(pageToBeInsertedId);
+    const pageToBeInsertedEl = pageToBeInserted.element;
+
+    const currentPage = this.getPageById_(currentPageId);
+    const currentPageEl = currentPage.element;
+
+    const nextPageId = currentPage
+        .getNextPageId(false /*opt_isAutomaticAdvance */);
+
+    currentPageEl.setAttribute(ADVANCE_TO_ATTR, pageToBeInsertedId);
```

Comment:
I think you will need to set `auto-advance-to` as well, in case the current page is using auto-advance.

Reply:
done